server/appengine/src/main/java/who/RefreshCaseStatsServlet.java

```diff
@@ -67,18 +70,106 @@ private static void updateData(
       snapshot.totalCases = 0L;
       snapshot.totalDeaths = 0L;
       data.snapshots.put(timestamp, snapshot);
+    } else {
+      // There shouldn't be a snapshot already except for GlobalStats!!!
```

Comment:
Pass in global as a boolean and then produce error if this codepath is encountered for non global stats? It would mean that we have a repeated timestamp which is likely a serious problem we should now about. Fine to defer to another PR.

Reply:
this was part of an earlier experiment.